### PR TITLE
Switch to NavigationExperimental

### PR DESCRIPTION
**Pull request by ericvicenti**
- Create a bunch of navigation actions
- Persist navigation state in redux store.navigation.openModals
- Initial "LoginScreen" is part of the main stack as part of the initial state
- Put nav logic into reducer
- Switch F8Navigator from Navigator to NavigationAnimatedView
- Test on iOS and Android

Known problems:
- Bad states when tapping away from android drawers instead of using back button
- Untested push notifs

I'd greatly appreciate some support from the community to review and help test the code!

### CODE REVIEW

File: js/F8Navigator.js [hunk 1]
```diff
@@ -74,95 +76,92 @@ var F8Navigator = React.createClass({
...
+    if (sceneState.type === 'ShareSettings') {
+      isVertical = false;
     }
-    if (route.session) {
+    return (
+      <F8NavigationCard
+        isVertical={isVertical}
+        renderScene={this.renderInnerScene}
+        {...props}>
+        {this.renderInnerScene(props)}
```
Comment (BlooJeans) [inline js/F8Navigator.js]: It looks like the {this.renderInnerScene(props)} doesn't get used anywhere.  F8NavigationCard uses this.props.renderScene to render the scene, and doesn't use this.props.children anywhere

**Comment by chirag04** [comment]
@ericvicenti is there a way to mimic navigators `FadeAndroid` as the default animation for push on android?

File: js/F8Navigator.js (hunk 1 above)
Comment (ericvicenti) [inline js/F8Navigator.js]: Good note! I'll remove that

File: js/F8NavigationCardStyleInterpolator.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    outputRange: [1, 1, 0.95],
+  });
+
+  const translateX = 0;
+  const translateY = position.interpolate({
+    inputRange,
+    outputRange: [height, 0, -10],
+  });
+
+  return {
```
Comment (ericvicenti) [inline js/F8NavigationCardStyleInterpolator.js]: @chirag04, here is where we might customize the animation for Android. But I'm not sure what the ideal settings are, so maybe we can iterate on it later?

Comment (chirag04) [inline js/F8NavigationCardStyleInterpolator.js]: Sounds good

**Comment by fingermark** [comment]
Does this patch still work?  Just trying to figure out why it wasn't merged, as I'm interested in learning how to integrate NavigationExperimental into my own project.

**Comment by ericvicenti** [comment]
Closing this because now we would want to use [React Navigation](https://reactnavigation.org/) instead

**Pull request closed without merging**